core/src/main/java/com/homeaway/streamingplatform/extensions/validation/SchemaRegistrar.java

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2018 Expedia Group.
+ * All rights reserved.  http://www.homeaway.com
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *      http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.homeaway.streamingplatform.extensions.validation;
+
+import java.util.Map;
+
+import com.homeaway.streamingplatform.exceptions.SchemaRegistrationException;
+import com.homeaway.streamingplatform.model.Stream;
+
+/**
+ * Interface for managing schema validation, registration
+ */
+public interface SchemaRegistrar {
+
+    /**
+     * Used to validate the stream's key and value schemas
+     *
+     * @param stream object
+     * @return whether schemas are valid or not
+     */
+    boolean isSchemaValid(Stream stream);
```

Comment:
Can we just pass the schema as string in the parameter.

also, it is preferred to use the same/similar signature or method names of the confluent's SchemaRegistry client
https://github.com/confluentinc/schema-registry/blob/master/client/src/main/java/io/confluent/kafka/schemaregistry/client/SchemaRegistryClient.java

Reply:
My thought is that we are not assuming the use of Confluent's schema registry which is why we are creating the abstraction in the first place.  Two simple methods make more sense to me.  @vinayakponangi what do you think?

Reply:
Hmm... @westeras can we use this as an interface?
https://homeaway.github.io/stream-registry/architecture/#schema-manager-interface

If not, help me understand the current differences and why.

Reply:
@neoword yes...I didn't know that existed...

Reply:
@neoword 
myself and Adam had discussion on this. 
so, the Interface `SchemaManager` at this point would have only 2 methods
```
public SchemaReference registerSchema(String subject, String schema)
                                        throws SchemaManagerException;
```
```
public boolean checkCompatibility(String subject, int version,
                                    String schema)
                                    throws SchemaException;```
Once we have use-cases for other methods, we would include the rest of the methods, and implement them.

Reply:
@neoword @prabhakar1983 another thing I just thought about: if we follow this contract and only provide subject/version/schema to these methods, we still need some mechanism for identifying the correct schema registry URL.  My original design passed the entire stream object into the schema manager interface, which had an infra manager instance, and then used the `vpcList` and `hint` from the stream to get the property URL from the infra manager.  I can think of two options to get around this: assume only one master schema registry URL, or provide a method via schema manager interface to lookup proper schema registry URL

Reply:
My only point here, is lets not re-invent the wheel.  

The interface proposed above was shamelessly pulled from here:
https://docs.confluent.io/current/schema-registry/docs/api.html

I believe most (all?) use cases are covered by that.
If you can think of a specific use case not covered by that, lets talk offline.